lib/utils/watchers.js

```diff
@@ -168,11 +173,11 @@ function handleRedditEvent({ target, detail: { type, data } }) {
 export function initObservers() {
 	if (isAppType('d2x')) {
 		document.addEventListener('reddit', (handleRedditEvent: any), true);
-		document.dispatchEvent(new CustomEvent('reddit.ready', {
-			detail: {
-				name: JSAPI_CONSUMER_NAME,
-			},
-		}));
+		const meta = document.createElement('meta');
```

Comment:
Firefox throws on getting properties from an object created by an extension. Probably security. So, new JSAPI handshake: move that metadata into the DOM and light a flare for the JSAPI to see.